pkg/kubernetes/version.go

```diff
@@ -0,0 +1,220 @@
+package kubernetes
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const (
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/
+	AnnotationSpinnakerArtifactVersion = `artifact.spinnaker.io/version`
+	AnnotationSpinnakerMonikerSequence = `moniker.spinnaker.io/sequence`
+	Annotation
+)
+
+type SpinnakerVersion struct {
+	Long  string
+	Short string
+}
+
+// AddSpinnakerVersionAnnotations adds the `artifact.spinnaker.io/version` and `moniker.spinnaker.io/sequence` annotations to the manifest to identify the version number of that resource
+func (c *controller) AddSpinnakerVersionAnnotations(u *unstructured.Unstructured, version SpinnakerVersion) error {
+	var err error
+
+	annotate(u, AnnotationSpinnakerArtifactVersion, version.Long)
+	annotate(u, AnnotationSpinnakerMonikerSequence, version.Short)
+
+	gvk := u.GroupVersionKind()
+
+	if strings.EqualFold(gvk.Kind, "deployment") {
+		d := NewDeployment(u.Object)
+
+		d.AnnotateTemplate(AnnotationSpinnakerArtifactVersion, version.Long)
+		d.AnnotateTemplate(AnnotationSpinnakerMonikerSequence, version.Short)
+
+		*u, err = d.ToUnstructured()
+		if err != nil {
+			return err
+		}
+	}
+
+	if strings.EqualFold(gvk.Kind, "replicaset") {
+		rs := NewReplicaSet(u.Object)
+
+		rs.AnnotateTemplate(AnnotationSpinnakerArtifactVersion, version.Long)
+		rs.AnnotateTemplate(AnnotationSpinnakerMonikerSequence, version.Short)
+
+		*u, err = rs.ToUnstructured()
+		if err != nil {
+			return err
+		}
+	}
+
+	if strings.EqualFold(gvk.Kind, "daemonset") {
+		ds := NewReplicaSet(u.Object)
+
+		ds.AnnotateTemplate(AnnotationSpinnakerArtifactVersion, version.Long)
+		ds.AnnotateTemplate(AnnotationSpinnakerMonikerSequence, version.Short)
+
+		*u, err = ds.ToUnstructured()
+		if err != nil {
+			return err
+		}
+	}
+
+	if strings.EqualFold(gvk.Kind, "statefulset") {
+		ss := NewStatefulSet(u.Object)
+
+		ss.AnnotateTemplate(AnnotationSpinnakerArtifactVersion, version.Long)
+		ss.AnnotateTemplate(AnnotationSpinnakerMonikerSequence, version.Short)
+
+		*u, err = ss.ToUnstructured()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// AddSpinnakerVersionLabels adds the `moniker.spinnaker.io/sequence` label to the manifest to identify the version number of that resource
+func (c *controller) AddSpinnakerVersionLabels(u *unstructured.Unstructured, version SpinnakerVersion) error {
```

Comment:
Move this function to the `label.go` file, next to the other functions that add labels.

Reply:
done